File: 1 Two sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {}
        for i, num in enumerate(nums):
            if target - num in hashMap:
                return [hashMap[target - num], i]
            else:
                hashMap[num] = i
        return

File: 1 two sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var map = new Dictionary<int, int>();
        
        for (int i = 0; i < nums.Length; i++) {
            int complement = target - nums[i];
            
            if (map.ContainsKey(complement))
                return [map[complement], i];
            
            map[nums[i]] = i;
        }
        return null; 
    }
}
